### PR TITLE
rip out the last bits of coffee-script

### PR DESCRIPTION
**Pull request by rossipedia**
big thanks to:

* [decaffeinate](http://decaffeinate-project.org/)
* [prettier-eslint](https://github.com/prettier/prettier-eslint-cli)

this is all the work in #578 squashed and rebased onto `master`

### CODE REVIEW

File: test/integration/instance-lookup-test.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      instanceName: 'badInstanceName',
+      timeout: 100,
+      retries: 1
+    },
+    callback
+  );
+};
+
+exports.badServer = function(test) {
+  // var config = getConfig();
```
Comment (tvrprasad) [inline test/integration/instance-lookup-test.js]: You can remove this line.

File: test/integration/connection-test.js [hunk 2]
```diff
@@ -0,0 +1,1265 @@
+var config, connection;
```
Comment (tvrprasad) [inline test/integration/connection-test.js]: Can you remove that line?

File: test/integration/connection-test.js [hunk 3]
```diff
@@ -0,0 +1,1265 @@
...
+var async = require('async');
+var Connection = require('../../src/connection');
+var Request = require('../../src/request');
+var fs = require('fs');
+
+/* eslint-disable no-unused-vars */
```
Comment (tvrprasad) [inline test/integration/connection-test.js]: Can you remove that and make it work?

File: test/integration/connection-test.js [hunk 4]
```diff
@@ -0,0 +1,1265 @@
...
+  connection.on('errorMessage', function(error) {
+    //console.log("#{error.number} : #{error.message}")
+  });
+
+  connection.on('debug', function(text) {
+    //console.log(text)
+  });
+};
+
+exports.execSqlMultipleTimes2 = function(test) {
```
Comment (tvrprasad) [inline test/integration/connection-test.js]: You can remove this test. This looks like a less complete duplicate of execSqlMultipleTimes. In the .coffee file the same name is used twice, so one of them is overriding the other. The conversion tool seems to have renamed the second instance.

File: test/unit/message-io-test.js [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+exports.receiveMultiplePacketsWithMoreThanOnePacketFromOneChunk = function(
+  test
+) {
+  test.expect(1);
+
+  var payload = new Buffer([1, 2, 3, 4, 5, 6]);
+  // var payload1 = payload.slice(0, 2);
+  // var payload2 = payload.slice(2, 4);
+  // var payload3 = payload.slice(4, 6);
```
Comment (tvrprasad) [inline test/unit/message-io-test.js]: You can remove these commented lines.

File: test/unit/token/row-token-parser-test.js [hunk 6]
```diff
@@ -0,0 +1,981 @@
...
+  buffer.writeString(value.slice(0, 3));
+  buffer.writeUInt32LE(3);
+  buffer.writeString(value.slice(3, 6));
+  buffer.writeUInt32LE(0);
+  //console.log(buffer.data)
+
+  try {
+    var parser = new Parser({ token() {} }, colMetaData, options);
+    parser.write(buffer.data);
+    /* eslint-disable */
```
Comment (tvrprasad) [inline test/unit/token/row-token-parser-test.js]: You can get rid of this by changing:
<code>var token = parser.read();</code>
to
<code>parser.read();</code>

File: test/unit/tracking-buffer/bigint-test.js [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+    0xfb,
+    0xff,
+    0xff
+  ]);
+
+  test.done();
+};
+
+var assertBuffer = function(test, actual, expected) {
+  return (() => {
```
Comment (tvrprasad) [inline test/unit/tracking-buffer/bigint-test.js]: I think this code can be simplified:
```JavaScript
var assertBuffer = function (test, actual, expected) {
  for (var i = 0, end = actual.length - 1; i <= end; i++) {
    if (actual[i] !== expected[i]) {
      console.log('actual  ', actual);
      console.log('expected', new Buffer(expected));
      test.ok(false);
    }
  }
};
```

**Comment by rossipedia** [comment]
Yeah I can take care of this stuff tomorrow. Thanks for the thorough review :)

**Comment by arthurschreiber** [comment]
@rossipedia This is awesome! 😍

I sliced this PR into smaller commits and changed the commit messages to make them `semantic-release` friendly, but ensured that authorship of these commits is still attributed to you!

Thank you again for working on this!

**Comment by rossipedia** [comment]
Awesome! Always happy to rid the world of a little bit more coffee-script ;)

**Pull request merged**